compose/cli/main.py

```diff
@@ -296,6 +297,57 @@ def json_format_event(event):
             print(formatter(event))
             sys.stdout.flush()
 
+    def exec_command(self, project, options):
+        """
+        Execute a command in a running container
+
+        Usage: exec [options] SERVICE COMMAND [ARGS...]
+
+        Options:
+            -d                Detached mode: Run command in the background.
+            --privileged      Give extended privileges to the process.
+            --user USER       Run the command as this user.
+            -T                Disable pseudo-tty allocation. By default `docker-compose exec`
+                              allocates a TTY.
+            --index=index     index of the container if there are multiple
+                              instances of a service [default: 1]
+        """
+        index = int(options.get('--index'))
+        service = project.get_service(options['SERVICE'])
+        try:
+            container = service.get_container(number=index)
+        except ValueError as e:
+            raise UserError(str(e))
+        command = [options['COMMAND']] + options['ARGS']
+        tty = not options["-T"]
+
+        create_exec_options = {
+            "privileged": options["--privileged"],
+            "user": options["--user"],
+            "tty": tty,
+            "stdin": tty,
+        }
+
+        exec_id = container.create_exec(command, **create_exec_options)
+
+        if options['-d']:
+            container.start_exec(exec_id, tty=tty)
+            return
+
+        try:
+            operation = ExecOperation(
+                    project.client,
+                    exec_id,
+                    interactive=tty,
+            )
+            pty = PseudoTerminal(project.client, operation)
+            pty.start()
+        except signals.ShutdownException:
```

Comment:
This is only raised if you register the signal handlers, but I don't see that happening here.